### ISSUE TITLE
extern definition for measureText should return non-null

### ISSUE THREAD

**Issue by myphysicslab** [google/closure-compiler#2425]
In the extern [html5.js](https://github.com/google/closure-compiler/blob/master/externs/browser/html5.js#L550) the return type should be non-null.  Add an exclamation point like this:

```javascript
/**
 * @param {string} text
 * @return {!TextMetrics}
 * @nosideeffects
 */
CanvasRenderingContext2D.prototype.measureText = function(text) {};
```

The [spec for measureText](https://html.spec.whatwg.org/multipage/scripting.html#dom-context-2d-measuretext) makes no mention of possibly returning null.  

This change helps avoid needing code like this:

```javascript
var textMetrics = context.measureText(s);
if (!textMetrics) {
 throw new Error();
}
var textWidth = textMetrics.width;
```
instead you can write
```javascript
var textWidth = context.measureText(s).width;
```

**Comment by myphysicslab** [comment]
Same situation for [createLinearGradient](https://html.spec.whatwg.org/multipage/scripting.html#dom-context-2d-createlineargradient) and probably several other calls on `CanvasRenderingContext2D` such as `createRadialGradient` and  `createPattern`.

**Comment by brad4d** [comment]
@myphysicslab 
Fortunately, fixing return values to be non-nullable should be an easy change to land.
Care to do a pull request?

**Comment by myphysicslab** [comment]
As discussed on #860 I've also changed some parameter types to be non-nullable which can break existing code. Shall I remove those for now?  I think it would be good to have those definitions sooner, before more code is written to the looser incorrect specification.

**Comment by myphysicslab** [comment]
After thinking it over, I'll modify the pull request to only change return values to be non-nullable, not parameter values.  The question about non-nullable parameter values should be done on a separate issue and pull request.

**Comment by myphysicslab** [comment]
Fixed by #2435
